shtdlib.sh

```diff
@@ -227,17 +227,21 @@ function version_sort {
 
 # Returns the index number of the lowest version, in effect this means it
 # returns true if the first value is the smallest but will always return
-# the index of the lowest version
+# the index of the lowest version. In the case of multiple matches, the lowest
+# (the first match) index is returned.
 # Example:
 # compare_versions '1.1.1 1.2.2test' -> returns 0 # True
 # compare_versions '1.2.2 1.1.1' -> returns 1 # False
 # compare_versions '1.0.0 1.1.1 2.2.2' -> returns 0 # True
-# compare_versions '4.0.0 3.0.0 2.0.0 1.1.1test 1.0.0' -> returns 4 # False but
-# also position
+# compare_versions '4.0.0 3.0.0 2.0.0 1.1.1test 1.0.0 v5.0' -> returns 4 (False
```

Comment:
How about:

compare_versions '4.0.0 3.0.0 2.0.0 1.1.1test 1.0.0 v5.0' -> returns 4 (the index number which also evaluates as False in bash since any number other than 0 is Falsy).

Reply:
Updated (slightly modified to fit 80chars)